crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -185,6 +185,14 @@ impl Persistence {
             .ok_or(error::Auction::Missing)
             .map(|scores| (scores.block_deadline as u64).into())?;
 
+        if database::settlements::already_processed(&mut ex, auction_id)
```

Comment:
What's our policy regarding the DB operation time metric? Should we cover all the queries?

Reply:
Commented here when I added `get_auction` function: https://github.com/cowprotocol/services/pull/2734#discussion_r1682775716